packages/storage-ui/src/Contexts/PosthogContext.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react"
+import posthog from "posthog-js"
+import { Button, Typography, useLocation } from "@chainsafe/common-components"
+import { createStyles, makeStyles } from "@chainsafe/common-theme"
+import { Trans } from "@lingui/macro"
+import { useLocalStorage } from "@chainsafe/browser-storage-hooks"
+import { CSSTheme } from "../Themes/types"
+import { useUser } from "./UserContext"
+
+export type PosthogContext = {
+  hasOptedIn: boolean
+  posthogInitialized: boolean
+  captureEvent: (eventName: string, properties?: posthog.Properties) => void
+}
+
+type PosthogProviderProps = posthog.Config & {
+  children: React.ReactNode | React.ReactNode[]
+}
+
+const PosthogContext = React.createContext<PosthogContext>({
+  hasOptedIn: false,
+  posthogInitialized: false,
+  captureEvent: () => undefined
+})
+
+const useStyles = makeStyles(
+  ({ palette, breakpoints, constants, zIndex }: CSSTheme) => {
+    return createStyles({
+      cookieBanner: {
+        position: "fixed",
+        bottom: 0,
+        width: "100%",
+        display: "flex",
+        color: palette.common.white.main,
+        flexDirection: "column",
+        backgroundColor: constants.cookieBanner.backgroundColor,
+        padding: "16px 32px",
+        zIndex: zIndex?.layer1,
+        [breakpoints.down("sm")]: {
+          padding: "16px 16px"
+        }
+      },
+      bannerHeading: {
+        fontSize: 24,
+        lineHeight: "42px",
+        [breakpoints.down("sm")]: {
+          fontSize: 22,
+          lineHeight: "40px"
+        }
+      },
+      bannerText: {
+        fontSize: 14,
+        lineHeight: "18px",
+        marginBottom: constants.generalUnit * 1.5,
+        [breakpoints.down("sm")]: {
+          fontSize: 13,
+          lineHeight: "16px"
+        }
+      },
+      link: {
+        color: palette.common.white.main,
+        paddingLeft: constants.generalUnit
+      },
+      buttonSection: {
+        display: "flex",
+        flexDirection: "row",
+        margin: `${constants.generalUnit}px 0`
+      },
+      acceptButton: {
+        marginLeft: constants.generalUnit * 2
+      }
+    })
+  }
+)
+
+const TOUCHED_COOKIE_BANNER_KEY = "csf.touchedCookieBanner"
```

Comment:
So that we have 2 different approval
```suggestion
const TOUCHED_COOKIE_BANNER_KEY = "css.touchedCookieBanner"
```